src/error.rs

```diff
@@ -1,56 +1,87 @@
-use std::error::Error as StdError;
-use std::io::{Error as IOError, ErrorKind};
-
 #[derive(Debug, PartialEq, Clone, Eq, Display)]
 pub enum Error {
-    #[display(fmt = "parse error")]
-    ParseError,
-    #[display(fmt = "unaligned page access")]
-    Unaligned,
-    #[display(fmt = "out of bound access")]
-    OutOfBound,
-    #[display(fmt = "max cycles exceeded")]
-    InvalidCycles,
-    #[display(fmt = "cycles overflow")]
+    #[display(fmt = "aot error: dynasm ret {}", "_0")]
+    AotDynasm(i32),
+    #[display(fmt = "aot error: section is empty")]
+    AotSectionIsEmpty,
+    #[display(fmt = "aot error: section overlaps with another")]
+    AotSectionOverlaps,
+    #[display(fmt = "aot error: limit reached maximum dummy sections")]
+    AotLimitReachedMaximumDummySections,
+    #[display(fmt = "aot error: limit reached maximum labels")]
+    AotLimitReachedMaximumLabels,
+    #[display(fmt = "aot error: limit reached maximum sections")]
+    AotLimitReachedMaximumSections,
+    #[display(fmt = "aot error: limit reached maximum temp register")]
+    AotLimitReachedMaximumTempRegisters,
+    #[display(fmt = "aot error: out of bound due to not start of basic block")]
+    AotOutOfBoundDueToNotStartOfBasicBlock,
+    #[display(fmt = "asm error: {}", "_0")]
+    Asm(u8),
```

Comment:
Where is this error used?

Reply:
https://github.com/mohanson/ckb-vm/blob/error/src/machine/asm/mod.rs#L537

The `Asm` error will not be triggered unless there are bugs in the asm code